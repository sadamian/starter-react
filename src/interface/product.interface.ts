export type ProductCategory = "general" | "tech" | "home" | null;

export interface IProduct {
  readonly name: string;
  readonly price: number;
  readonly description: string;
  readonly image: string;
  readonly promotion?: boolean;
  readonly category?: ProductCategory;
}

export interface ImageProps {
  readonly image: string;
}

export interface TitleProps {
  readonly name: string;
}
