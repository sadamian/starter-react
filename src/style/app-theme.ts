/**
 * @file Theme of the App
 * @description
 * All Styled Components can access to this theme
 */

import { DefaultTheme } from "styled-components";

const appTheme: DefaultTheme = {
  borderRadius: "5px",

  colors: {
    main: "cyan",
    secondary: "magenta",
  },
  background: {
    color: "#f5f5f5",
  },
};

export { appTheme };
