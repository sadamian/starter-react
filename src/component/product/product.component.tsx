import React, { ReactNode } from "react";
import {
  Product_StyledComponent,
  ProductImage_StyledComponent,
  ProductTitle_StyledComponent,
} from "./product.styled.component";
import { ImageProps, IProduct, TitleProps } from "interface/product.interface";

export interface Props {
  product: IProduct;
}

const Product: React.FC<Props> = ({ product }) => {
  return (
    <Product_StyledComponent>
      <div style={{ display: "flex" }}>
        <ProductImage image={product.image} />
        <ProductTitle name={product.name} />
      </div>
    </Product_StyledComponent>
  );
};

export const ProductImage = ({ image }: ImageProps) => (
  <ProductImage_StyledComponent image={image} />
);

export const ProductTitle = ({ name }: TitleProps) => (
  <ProductTitle_StyledComponent>{name}</ProductTitle_StyledComponent>
);

export default Product;
