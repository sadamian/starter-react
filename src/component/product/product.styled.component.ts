import styled from "styled-components";
import toMaterialStyle from "material-color-hash";
import { ImageProps, TitleProps } from "interface/product.interface";

export const Product_StyledComponent = styled.div`
  border-radius: ${(props) => props.theme.borderRadius};
  background-color: ${(props) => props.theme.background.color};
  width: 150px;
  height: 50px;
`;

export const ProductTitle_StyledComponent = styled.div`
  color: ${(props) => props.theme.colors.main};
`;

export const ProductImage_StyledComponent = styled.div<ImageProps>`
  width: 50px;
  height: 50px;
  border-radius: 25px;
  background-color: ${(props) =>
    toMaterialStyle(props.image, 500).backgroundColor};
`;

export const ProductCategory_StyledComponent = styled.div`
  background-color: lol;
`;

export const ProductDescription_StyledComponent = styled.div`
  color: ${(props) => props.theme.colors.second};
`;

export const ProductPrice_StyledComponent = styled.div`
  color: ${(props) => props.theme.colors.main};
`;
