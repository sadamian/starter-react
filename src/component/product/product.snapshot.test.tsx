import React from "react";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";

import Product from "component/product/product.component";
import { IProduct } from "interface/product.interface";
import { appTheme } from "style/app-theme";

const product: IProduct = {
  name: "productos",
  category: "tech",
  description: "lelele",
  image: "jeje",
  price: 12,
};
describe("Snapshot Product Component", () => {
  test("Snap Product Properly", () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={appTheme}>
          <Product product={product} />
        </ThemeProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
