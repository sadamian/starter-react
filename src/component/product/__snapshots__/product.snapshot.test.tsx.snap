// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Product Component Snap Product Properly 1`] = `
<div
  className="sc-bdfBwQ NALLI"
>
  <div
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <div
      className="sc-dlfnbm ynGPZ"
    />
    <div
      className="sc-gsTCUz fYVHqO"
    >
      productos
    </div>
  </div>
</div>
`;
